Add reverse-proxy troubleshooting section to Kubernetes docs (#98)
packagist/private-packagist-docs: docs/self-hosted/kubernetes-troubleshooting.md

> a year. As Private Packagist only allows a time-drift of up to one (1) minute, we
> recommend using TOTP devices that have the ability to stay synchronized with
> the correct time (such as a phone, or re-programmable TOTP hardware devices).

#### Issues with Reverse-Proxy running in front of the Kubernetes Cluster

Please follow the instructions below, if you are experiencing problems with the reverse-proxy not being able to connect to
the cluster and encountering errors like this:
```
Peer closed connection in SSL handshake (104: Connection reset by peer) while SSL handshaking to upstream
```

Ensure that the SNI (Server Name Indication) TLS Extension is properly set for requests to the ingress controller of the Kubernetes Cluster.

> This is not the case when using IPs in `proxy_pass` on the reverse proxy and will result in an SSL handshake error.

Reviewer: @pscheit Also pinging you to get your opinion here :)

> This is not the case when using IPs in `proxy_pass` on the reverse proxy and will result in an SSL handshake error.

Reviewer: Note, alternatively we just move the last sentence to here and mention that the "following examples assume you are using nginx, please consult the documentation of other reverse-proxy servers to achieve the same result.", something like that. Just something to avoid confusion :)

> This is not the case when using IPs in `proxy_pass` on the reverse proxy and will result in an SSL handshake error.

Reviewer 2: the last suggestion is better, because now the two sentences basically tell us the same story?

> a year. As Private Packagist only allows a time-drift of up to one (1) minute, we
> recommend using TOTP devices that have the ability to stay synchronized with
> the correct time (such as a phone, or re-programmable TOTP hardware devices).

#### Issues with Reverse-Proxy running in front of the Kubernetes Cluster

Please follow the instructions below, if you are experiencing problems with the reverse-proxy not being able to connect to
the cluster and encountering errors like this:
```
Peer closed connection in SSL handshake (104: Connection reset by peer) while SSL handshaking to upstream
```

> Ensure that the SNI (Server Name Indication) TLS Extension is properly passed in requests to the cluster
> for SNI to work correctly on the ingress. When using IPs as upstream hostnames on the reverse-proxy, this is not the case

Reviewer 2: ```suggestion
Ensure that the SNI (Server Name Indication) TLS Extension is properly set for requests to the ingress controller of the Kubernetes Cluster.
```

Reviewer 2: Maybe? Not sure

Author: Sounds good 👍

> for SNI to work correctly on the ingress. When using IPs as upstream hostnames on the reverse-proxy, this is not the case
> by default and will result in a certificate error.

Reviewer 2: ```suggestion
for SNI to work correctly on the ingress. This is not the case when using IPs in `proxy_pass` on the reverse proxy and will result in an SSL handshake error.
```

Reviewer 2: This makes it probably a bit easier to read? Not sure about the exact wording

> To pass the SNI hostname from the incoming request to the upstream server, apply the following settings when using
> NGINX as a reverse-proxy:

Reviewer 2: ```suggestion
To pass the SNI hostname from the incoming request to the upstream server, add the following directives to NGINX:
```

Reviewer 2: are they called directives?

Reviewer: Directive is the right term indeed (https://nginx.org/en/docs/dirindex.html)
